react/components/AutocompleteInput/Option.tsx

```diff
@@ -0,0 +1,91 @@
+import classnames from 'classnames'
+import PropTypes from 'prop-types'
+import React, { useState } from 'react'
+
+/** Structured Autocomplete option shape */
+const structuredAutocompleteOptionShape = PropTypes.shape({
+  value: PropTypes.any,
+  label: PropTypes.string.isRequired,
+})
+
+type StructuredAutocompleteOption = PropTypes.InferType<
+  typeof structuredAutocompleteOptionShape
+>
+
+export type AutocompleteOption = string | StructuredAutocompleteOption
+
+/** Autocomplete option shape */
+export const autocompleteOptionShape = PropTypes.oneOfType([
+  PropTypes.string,
+  structuredAutocompleteOptionShape,
+])
+ 
+export const getTermFromOption = (option: AutocompleteOption): string =>
+  typeof option === 'string' ? option : option.label
+
+const propTypes = {
+  /** Determine if the option should have a rounded bottom */
+  roundedBottom: PropTypes.bool,
```

Comment:
Is that a UI feature?

Reply:
Yes. It can be renamed to `lastOption`, if you think it's better...